.github/workflows/test_build.yml

```diff
@@ -1,34 +1,84 @@
+      - name: Check out repository
+        uses: actions/checkout@v3
+      - name: Set up conda
+        uses: conda-incubator/setup-miniconda@v2
         with:
+          auto-update-conda: true
+          miniconda-version: "latest"
+          activate-environment: build_binary
           python-version: ${{ matrix.python-version }}
-      - name: Check out repository
-        uses: actions/checkout@v2
+      - name: Check conda info
+        run: conda info
+      - name: Check python version with conda
+        run: python --version
+      - name: Update pip
+        run: python -m pip install --upgrade pip
+      - name: Update build/packaging tools
+        run: python -m pip install --upgrade setuptools wheel twine
       - name: Install dependencies
         run: |
-          pip3 install --pre torch torchvision torchtext --extra-index-url https://download.pytorch.org/whl/nightly/cpu
-          pip3 install -r dev-requirements.txt
-      - name: Build wheel
+          python -m pip install --pre torch torchvision torchtext --extra-index-url \
+            https://download.pytorch.org/whl/nightly/cpu
+          python -m pip install -r requirements.txt
+      - name: Test installation of dependencies
         run: |
-          pip3 install wheel
-          python setup.py bdist_wheel
-      - name: Install wheel
+          python -c "import torch"
+          echo "Import torch succeeded"
+          python -c "import torchtext"
+          echo "Import torchtext succeeded"
+          python -c "import torchvision"
+          echo "Import torchvision succeeded"
+      - name: Build torchmultimodal nightly
         run: |
-          pip3 install dist/torchmultimodal*.whl
-      - name: Test import
+          rm -r dist || true
+          python setup.py bdist_wheel --package_name torchmultimodal-nightly \
+            --python-tag=${{ matrix.python-tag }}
+      - name: Install torchmultimodal nightly
+        run: python -m pip install dist/torchmultimodal*.whl
+      - name: Test torchmultimodal installation
+        # Go one-level up to avoid importing the source code
         run: |
+          pushd ../
           python -c "import torchmultimodal"
-      - name: Run unit tests for sanity
+          popd
+      - name: Run unit tests against the installation
+        # import-model MUST be `append` to give precedence to the installed package
+        # rather than the source code
+        run: |
+          python -m pip install pytest pytest-mock pytest-cov
+          pytest tests --import-mode=append --junitxml=junit/test-results.xml \
+            --cov-report=xml --cov-report=html --cov=. --durations=20 -vv
+      - name: Push wheel to PyPI
+        env:
+          PYPI_API_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
         run: |
-          pytest test -vv
+          python -m twine upload dist/torchmultimodal_nightly-*.whl \
+            --username __token__ --password "$PYPI_API_TOKEN" \
+            --skip-existing \
```

Comment:
what happens if we want to update same day or not possible?

Reply:
PyPI's convention is that we cannot override an uploaded wheel of the same name even if the previous one is bad.
The only way I am aware of is to re-name the updated `.whl` and do a manual upload from your local environment.